src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -763,7 +763,8 @@ public void ProcessTransaction(Transaction transaction, int? blockHeight = null,
                     // Check if the outputs contain one of our addresses.
                     if (this.keysLookup.TryGetValue(utxo.ScriptPubKey, out HdAddress pubKey))
                     {
-                        this.AddTransactionToWallet(transaction.ToHex(), hash, transaction.Time, transaction.IsCoinStake, transaction.Outputs.IndexOf(utxo), utxo.Value, utxo.ScriptPubKey, blockHeight, block);
+                        this.AddTransactionToWallet(transaction.ToHex(), hash, transaction.Time, transaction.IsCoinStake, transaction.Outputs.IndexOf(utxo),
+                            utxo.Value, utxo.ScriptPubKey, blockHeight, block, isPropagated ?? block != null);
```

Comment:
Following on from my previous comment, you can now remove the condition `isPropagated ?? block != null` and just use `isPropagated`.